sync_ci/pkg/detect/detect.go

```diff
@@ -0,0 +1,363 @@
+}
+
+func handleCaseIfHistoryExists(cfg model.Config, dbGithub *gorm.DB, issueNumStr string, repo string, caseName string, joblinks []string, issueCases []*model.CaseIssue, test bool) ([]*model.CaseIssue, error) {
+	stillValidIssues, err := dbGithub.Raw(model.CheckClosedTimeSql, issueNumStr, repo, searchIssueIntervalStr).Rows()
+	if err != nil {
+		return nil, err
+	}
+	if !stillValidIssues.Next() {
+		issueCase := model.CaseIssue{
+			IssueNo:   0,
+			Repo:      repo,
+			IssueLink: sql.NullString{},
+			Case:      sql.NullString{caseName, true},
+			JobLink:   sql.NullString{joblinks[0], true},
+		}
+		issueCases = append(issueCases, &issueCase)
+	} else { // mention existing issue
+		var url string
+		err = stillValidIssues.Scan(&url)
+		if err != nil {
+			log.S().Error("failed to extract existing issue url", err)
+		}
+
+		issueId := strings.Split(url, "/issues/")[1]
+		err = MentionIssue(cfg, repo, issueId, joblinks[0], test)
+		if err != nil {
+			log.S().Error(err)
+		}
+	}
+	return issueCases, nil
+}
+
+func GetNightlyCases(cfg model.Config, filterStartTime, now time.Time, test bool) ([]*model.CaseIssue, error) {
+	cidb, err := SetupDB(cfg.Dsn)
+	if err != nil {
+		return nil, err
+	}
+	ghdb, err := SetupDB(cfg.GithubDsn)
+	if err != nil {
+		return nil, err
+	}
+	csdb, err := SetupDB(cfg.CaseDsn)
+	if err != nil {
+		return nil, err
+	}
+	rows, err := cidb.Raw(model.GetCINightlyCase, formatT(filterStartTime), formatT(now)).Rows()
+	RepoNightlyCase := map[string]map[string] []string {}
+	issueCases := []*model.CaseIssue{}
+
+	if err != nil {
+		return nil, err
+	}
+	for rows.Next() {
+		var rawCase []byte
+		var cases []string
+		var jobid string
+		var job string
+		err = rows.Scan(&rawCase, &jobid, &job)
+		if err != nil {
+			log.S().Error(err)
+			continue
+		}
+		err = json.Unmarshal(rawCase, &cases)
+		if err != nil {
+			log.S().Error(err)
+			continue
+		}
+		repo := extractRepoFromJobName(job)
+		if repo == "others" { // mixed unknown repos
+			continue
+		}
+
+		for _, c := range cases {
+			if _, ok := RepoNightlyCase[repo]; !ok {
+				RepoNightlyCase[repo] = map[string] []string{}
+			}
+			if _, ok := RepoNightlyCase[repo][c]; !ok {
+				link := fmt.Sprintf(baselink, job, jobid)
+				RepoNightlyCase[repo][c] = []string {link}
+
+				issueCase := model.CaseIssue{
+					IssueNo:   0,
+					Repo:      repo,
+					IssueLink: sql.NullString{},
+					Case:      sql.NullString{c, true},
+					JobLink:   sql.NullString{link, true},
+				}
+				issueCases = append(issueCases, &issueCase)
+			}
+		}
+	}
+	issueCases, err = handleCasesIfIssueExists(cfg, RepoNightlyCase, csdb, ghdb, test)
+	if err != nil {
+		log.S().Error("Failed to handle existing issue case", err)
+	}
+
+	return issueCases, nil
+}
+
+func extractRepoFromJobName(job string) string {
+	if strings.Contains(job, "tidb") {
+		return "pingcap/tidb"
+	}
+	if strings.Contains(job, "tikv") {
+		return "pingcap/tidb"
+	}
+	if strings.Contains(job, "pd") {
```

Comment:
use regrex `^pd` `^tikv_ghpr` `^tidb_ghpr` to match

Reply:
got it